### PR TITLE
Status page

### PR DESCRIPTION
**Pull request by Zoruk**
## Description

Add a web status wich work with multiple worker on same database

## How Has This Been Tested?

Tested with about 50 worker

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Add option

### CODE REVIEW

**Comment by justsomedudeonthenet** [comment]
I like the idea.  How does it handle multiple separate copies of the map running, like a beehive layout?

Many people were concerned when last talking about it that it should have password protection, as it reveals your usernames.

**Comment by Zerotask** [comment]
Yes, a password protection ist a must have for this.

**Comment by Zoruk** [comment]
@justsomedudeonthenet @Zerotask Password added

**Comment by aldarund** [comment]
So what about multiple copies like a beehive? Does it work with it?

**Comment by Zoruk** [comment]
@aldarund Yep when you lunch the app with -sn "XXX" it will appear with XXX as table header in the status page but each beehive must have a diverent name

**Comment by subzerofun** [comment]
Hey @Zoruk, thanks for the great addon! Would like to test it, but when i use this as my cmd args:

`python runserver.py -a ptc -u 'username' -p 'password' -st 10 -l 12,34 -D data.db -k 'mykey' --status-name 'STATUS_NAME' --status-page-password 'mypassword'`

i can open `localhost:5000/status` and the login page gets rendered, but after i put in the above defined password and click on login, the status page doesn´t get loaded... the browsers just refreshes and changes the url from **/status** to **/status?**. so for some reason i can´t get past the login screen. i tried to bypass the password query by changing your code, but that didn´t work either...

Is there something wrong with my arguments? Do i need to run more than one worker? I´m on OSX, tried Chrome, Firefox and Safari – everywhere with the same result.

When i look at the terminal output everything looks OK:
```
[ search-overseer][        search][    INFO] Search overseer starting
[ search-overseer][        search][    INFO] Starting status database thread
[ search-overseer][        search][    INFO] Starting search worker threads
```

**Comment by Zoruk** [comment]
The login is handled by javascript. The page should not reload

```javascript
$('#password_form').submit(function (event) {
  event.preventDefault()
  statusPagePassword = $('#password').val()
  ...
})
```

**Comment by ItsCEED** [comment]
Hi @Zoruk im kinda newbie, when you gonna make the documentation on how to use it? Also, if i run for example 2 workers x 5 screens(ports) do i get all the info on the main one (5000)?

**Comment by chuyskywalker** [comment]
I'm excited to see this get fleshed out.

Generally speaking, databases really aren't for logging...but...I also think this is more useful to have this feature, than to reject it under demand of a rewrite with a more complex "parent <-- child" api-esque pushing setup.

A quick, 30 second look over; you do need to audit the html page since it's including a boat load of stuff that I'm sure is unused.

I also would not recommend having the data delivered back via `raw_data` -- make a unique end point.

**Comment by Zoruk** [comment]
@chuyskywalker Done

**Comment by jnovack** [comment]
I don't like `/status`.  Everyone knows it is there.  It's just like `/admin`.

I RECOMMEND a basic `--status-page URLENCODED` so I can at least mask it.  Accept any URL Encoded string without `/`.  Examples: `status`, `secret_status`, `my-status-page` `0cba061c25c86424187f08b0344c13d1`

File: pogom/search.py [hunk 1]
```diff
@@ -17,27 +17,27 @@
...
 from pgoapi import utilities as util
 from pgoapi.exceptions import AuthException
+from pgoapi.utilities import f2i
+from queue import Queue, Empty
 
-from .models import parse_map, Pokemon
-from .fakePogoApi import FakePogoApi
 import terminalsize
+from .fakePogoApi import FakePogoApi
+from .models import parse_map, Pokemon, WorkerStatus, MainWorker
```
Comment (jnovack) [inline pogom/search.py]: What's with the reorganization?

**Comment by jnovack** [comment]
Secondarily, for the python dummies (myself included), does this work with out-of-process spawned workers?  Someone may have a use-case of runnning multiple searchers on multiple machines, feeding only.  Will this properly track them?

File: pogom/search.py (hunk 1 above)
Comment (Zoruk) [inline pogom/search.py]: Oops i just hit the refractoring of import if you look there is also the same removed

**Comment by Zoruk** [comment]
Yes it's work with multiples worker in multiples process on different machines

**Comment by Zerotask** [comment]
@jnovack you are a grumpy cat. I always read that you dislike things.

**Comment by jnovack** [comment]
Everyone thinks their patch is a work of art and perfect. (Mine included)  So with others, my strength is to come in, apply some happy little trees and make a "good patch" a "GREAT patch".  Make sure that someone considered other circumstances; ensure that a patch does not lead to a deviation of the project goals.  Just move the needle a bit.

The fact is, I rarely comment on PRs I hate, and I mostly comment and post on issues and PRs I *like* so they can be PRs I LOVE. 

Development is hard.  No Pain, No Gain
image

**Comment by Zerotask** [comment]
@Zoruk this tool is really amazing. 
I would suggest to add an accordion (jquery maybe) to switch between different views: a) all a) active b) banned / not working

At the moment I have big trouble with identifying banned accounts (to swap them). This would help a lot.

**Comment by mew905** [comment]
I'd actually like to expand this to encompass the radar as a whole and grab the info from a database. How would one go about it?

**Comment by maddhatter** [comment]
I _love_ this update. I needed this so badly, and this does such a better job than what I was thinking up...

The only thing that would make it better, IMHO, would be if whenever a worker starts up, it clears out old Worker statuses. That way if I notice a banned account and swap it out, I don't have to keep looking at the banned accounts' statuses on the page.

Something like this right at the beginning of `worker_status_db_thread` (before the `while True:`) seems to do it for me:
```python
log.info("Clearing previous statuses for '%s' worker", name)
WorkerStatus.delete().where(WorkerStatus.worker_name == name).execute()
```

File: static/js/status.js [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    <div class="status_table" id="table_${hash}">
+      <div class="status_row header">
+        <div class="status_cell">
+          Username
+        </div>
+        <div class="status_cell">
+          Success
+        </div>
+        <div class="status_cell">
+          Faill
```
Comment (maddhatter) [inline static/js/status.js]: Looks like there's a typo here

Comment (jnovack) [inline static/js/status.js]: Perhaps intentional?  Sarcasm?

**Comment by maddhatter** [comment]
@Zoruk  Would love to see this get merged. Sent you two PRs:

1. [#1](https://github.com/Zoruk/PokemonGo-Map/pull/1) to resolve the conflicts w/ develop
2. [#2](https://github.com/Zoruk/PokemonGo-Map/pull/2) to make the two tiny changes I suggested in my previous comments. (This one has conflicts, but they'll go away once [#1](https://github.com/Zoruk/PokemonGo-Map/pull/1) is merged)

**Comment by jnovack** [comment]
@maddhatter Zoruk has been AFK for 10 days.  Do you want to resubmit this with your fixes?

**Comment by maddhatter** [comment]
I've submitted #1090 to hopefully get this merged.

**Pull request closed without merging**